First/ViewController.swift

```diff
@@ -100,40 +94,46 @@ class ViewController: UIViewController {
             make.left.lessThanOrEqualTo(blackBackgroundView.snp.leftMargin).offset(kLeftMargin).priority(.high)
             make.right.greaterThanOrEqualTo(blackBackgroundView.snp.rightMargin).offset(-kRightMargin).priority(.high)
         }
         
         passwordTextField.snp.makeConstraints{make in
             make.height.equalTo(kElementsHeight).priority(.required)
-            make.top.lessThanOrEqualTo(userNameTextField.snp.bottomMargin).offset(kTopMargin).priority(.medium)
+            make.top.greaterThanOrEqualTo(userNameTextField.snp.bottomMargin).priority(.required)
+            make.top.lessThanOrEqualTo(userNameTextField.snp.bottomMargin).offset(kTopMargin).priority(.low)
             make.left.lessThanOrEqualTo(blackBackgroundView.snp.leftMargin).offset(kLeftMargin).priority(.high)
             make.right.greaterThanOrEqualTo(blackBackgroundView.snp.rightMargin).offset(-kRightMargin).priority(.high)
         }
         
         loginButton.snp.makeConstraints{make in
             make.height.equalTo(kElementsHeight).priority(.required)
-            make.top.lessThanOrEqualTo(passwordTextField.snp.bottomMargin).offset(kTopMargin).priority(.medium)
+            make.top.greaterThanOrEqualTo(passwordTextField.snp.bottomMargin).priority(.required)
+            make.top.lessThanOrEqualTo(passwordTextField.snp.bottomMargin).offset(kTopMargin).priority(.low)
             make.left.lessThanOrEqualTo(blackBackgroundView.snp.leftMargin).offset(kLeftMargin).priority(.high)
             make.right.greaterThanOrEqualTo(blackBackgroundView.snp.rightMargin).offset(-kRightMargin).priority(.high)
         }
         
         troubleButton.snp.makeConstraints{make in
             make.height.equalTo(kElementsHeight).priority(.required)
-            make.top.lessThanOrEqualTo(loginButton.snp.bottomMargin).offset(kTopMargin).priority(.medium)
+            make.top.greaterThanOrEqualTo(loginButton.snp.bottomMargin).priority(.required)
+            make.top.lessThanOrEqualTo(loginButton.snp.bottomMargin).offset(kTopMargin).priority(.low)
             make.left.lessThanOrEqualTo(blackBackgroundView.snp.leftMargin).offset(kLeftMargin).priority(.high)
             make.right.greaterThanOrEqualTo(blackBackgroundView.snp.rightMargin).offset(-kRightMargin).priority(.high)
-            make.bottom.greaterThanOrEqualTo(blackBackgroundView.snp.bottomMargin).offset(-kTroubleBottomMargin).priority(.required)
+            make.bottom.greaterThanOrEqualTo(blackBackgroundView.snp.bottomMargin).offset(-kTroubleBottomMargin).priority(.low)
+            make.bottom.lessThanOrEqualTo(blackBackgroundView.snp.bottomMargin).priority(.required)
         }
         
         legalTermsLabel.snp.makeConstraints{make in
-            make.top.greaterThanOrEqualTo(blackBackgroundView.snp.bottomMargin).offset(-kLegalTopMargin).priority(.required)
+            make.top.greaterThanOrEqualTo(blackBackgroundView.snp.bottom).priority(.required)
```

Comment:
w zasadzie wszystkie greater/lessThanOrEqualTo mozesz z powrotem zamienic na equalTo i wyrzucic priority, bo jest to czytelniejsze, z wyjatkiem tego, gdzie wydaje mi sie ze jak zostawisz cos w stylu greaterThanOrEqualTo(constant: 10) bedzie wszystko git